### DISCUSSION TITLE
dpd action for IKE

### DISCUSSION THREAD

**Discussion by psalty** [strongswan/strongswan#1589] [Q&A]
What would be the behavior of dpd_action for conn?

I see the following configuration for IKE but no information such as conn.dpd_action. (not child sa level)

<conn>.dpd_delay
<conn>.dpd_timeout

this might be useful for childless sa

**Comment by tobiasbrunner** [comment]
> What would be the behavior of dpd_action for conn?

Don't know, but there is currently no option for it.

> this might be useful for childless sa

What's the purpose of the childless IKE_SA in that case?

**Reply by psalty** [reply]
the Purpose is in the childless SA case, there will be IKE only.

IKE does DPD, but no action when DPD timed out as child SA does.

It would be nice to have such capablity.

**Reply by tobiasbrunner** [reply]
There are currently no plans for such a feature.
